lib/python/qmk/commands.py

```diff
@@ -52,13 +63,52 @@ def compile_configurator_json(user_keymap, bootloader=None):
 
         A command to run to compile and flash the C file.
     """
-    # Write the keymap C file
-    qmk.keymap.write(user_keymap['keyboard'], user_keymap['keymap'], user_keymap['layout'], user_keymap['layers'])
+    # Write the keymap.c file
+    keyboard_filesafe = user_keymap['keyboard'].replace('/', '_')
+    target = f'{keyboard_filesafe}_{user_keymap["keymap"]}'
+    keyboard_output = Path(f'{KEYBOARD_OUTPUT_PREFIX}{keyboard_filesafe}')
+    keymap_output = Path(f'{keyboard_output}_{user_keymap["keymap"]}')
+    c_text = qmk.keymap.generate_c(user_keymap['keyboard'], user_keymap['layout'], user_keymap['layers'])
+    keymap_dir = keymap_output / 'src'
+    keymap_c = keymap_dir / 'keymap.c'
+
+    keymap_dir.mkdir(exist_ok=True, parents=True)
+    keymap_c.write_text(c_text)
 
     # Return a command that can be run to make the keymap and flash if given
-    if bootloader is None:
-        return create_make_command(user_keymap['keyboard'], user_keymap['keymap'])
-    return create_make_command(user_keymap['keyboard'], user_keymap['keymap'], bootloader)
+    verbose = 'true' if cli.config.general.verbose else 'false'
+    color = 'true' if cli.config.general.color else 'false'
+    make_command = [
+        'make',
```

Comment:
can we get some sort of parallel build support here?

Reply:
added in bae2c9270